### DISCUSSION TITLE
Group By on  rest API [Jun 23]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2521] [Discord Threads]
> 🙋‍♂️ **Cris Acosta**

**Comment by o1lab** [comment]
> 🙋‍♂️ **Cris Acosta**  
#### Anyone?  
#  
> 🙋‍♂️ **wkw**  
#### Example:   
####    
#### Project Name: `Foo`   
#### Table Name: `Bar`   
#### View Name: `Bar`   
####    
#### `/api/v1/db/data/{orgs}/{projectName}/{tableName}/groupby`   
####     
#### ```bash   
#### curl --request GET \   
#### --url 'http://localhost:8080/api/v1/db/data/noco/Foo/Bar/groupby?column_name=Title' \   
#### --header 'xc-auth: <REDACTED>'   
#### ```   
####    
#### `/api/v1/db/data/{orgs}/{projectName}/{tableName}/views/{viewName}/groupby`   
####    
#### ```bash   
#### curl --request GET \   
#### --url 'http://localhost:8080/api/v1/db/data/noco/Foo/Bar/views/Bar/groupby?column_name=Title' \   
#### --header 'xc-auth: <REDACTED>'   
#### ```   
####    
#### Result:   
####    
#### ```bash   
#### {"list":[{"count":2,"Title":"A"},{"count":1,"Title":"B"},{"count":1,"Title":"C"}],"pageInfo":{"totalRows":4,"page":1,"pageSize":25,"isFirstPage":true,"isLastPage":true}}   
#### ```  
![image0](https://cdn.discordapp.com/attachments/986051928725262377/986207801921663016/unknown.png)  
#  
> 🙋‍♂️ **Cris Acosta**  
#### I´m go try!  
#  
> 🙋‍♂️ **Cris Acosta**  
#### Hi! Show only Count and defined column in the group by  clause  
#  
> 🙋‍♂️ **wkw**  
#### sorry. wt do u mean?  
#  
> 🙋‍♂️ **Cris Acosta**  
#### I need reproduce this in the Nocodb   
####    
#### SELECT    
#### usu.NomeUser AS nome,   
#### COUNT(cli.CodCliente) AS Total_de_Nome_RazaoSocial   
####    
#### FROM Atendimentos AS ate   
####    
####    
#### INNER JOIN Clientes cli ON ate.CodCliente =  cli.CodCliente    
#### INNER JOIN Usuarios usu ON ate.CodAgente =  usu.CodUser   
####    
####    
#### WHERE (ate.criadoEm)>="2022-06-01" And (ate.criadoEm)<=NOW()   
####    
#### GROUP BY cli.CodCliente, usu.NomeUser  
#  
> 💁 **o1lab**  
#### currently, joins are on primary and foreign key   
#### is that that case in the above query ?   
#### Easier way I feel is, you could create a view for this in your database   
#### and then do metadata-sync on nocodb  
#  
> 💁 **o1lab**  
#### you get the api  
#  
> 🙋‍♂️ **Cris Acosta**  
#### Hi Olla!  
#  
> 🙋‍♂️ **Cris Acosta**  
#### I made the joins like FK, ok  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/986051928725262377)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
